forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -138,6 +138,10 @@ public static boolean cardHasProperty(Card card, String property, Player sourceC
             if (source.hasChosenCard(card)) {
                 return false;
             }
+        } else if (property.startsWith("ChosenMode")) {
+            if (!card.getChosenMode().equals(property.substring(10))) {
```

Comment:
Probably need an extra check because ChosenModes are linked?

Or are we doing that way later with my LinkedAbility MR?

Reply:
yea, just transforming these scripts with statics will help later